src/app.ts

```diff
@@ -0,0 +1,95 @@
+﻿let grid: iq.onboard.Grid,
+	refreshHandle: number = 99;
+
+$(() => {
+	grid = new iq.onboard.Grid("#grid");
+
+	initialize();
+});
+
+function initialize() {
+	disableformSubmission();
+	initNavigationHandlers();
+	initControlPanel();
+	initResizeHandler();
+}
+
+function disableformSubmission() {
+	$("form").submit(() => {
+		return false;
+	});
+}
+
+function initNavigationHandlers() {
+	$("#next-button").click(() => {
+		grid.pageBy(1);
+		updatePageDisplay(grid.getPageNumber());
+	});
+
+	$("#prev-button").click(() => {
+		grid.pageBy(-1);
+		updatePageDisplay(grid.getPageNumber());
+	});
+
+	$("#first-button").click(() => {
+		grid.pageToFirst();
+		updatePageDisplay(grid.getPageNumber());
+	});
+
+	$("#last-button").click(() => {
+		grid.pageToLast();
+		updatePageDisplay(grid.getPageNumber());
+	});
+}
+
+function initControlPanel() {
+	initControlPanelAnimation();
+	initNavigationControlAnimation();
+	initPageSelector();
+}
+
+function initControlPanelAnimation() {
+	let $controlPanel = $(".control-panel");
```

Comment:
Using a CSS class selector can return multiple JQuery instances. If you had another DOM element re-using the same CSS class (which is likely because re-usability is the purpose of CSS classes), your event handler would be attached to both elements. 

We prefer using ID selectors when attaching event handlers unless you are absolutely sure that the CSS class won't be re-used elsewhere (but I don't ever take that chance)